workshops/painting_app/painting_app.md

```diff
@@ -0,0 +1,323 @@
+```js
+	var canvas = document.getElementById("paint-canvas");
+	var context = canvas.getContext("2d");
+	var boundings = canvas.getBoundingClientRect();
+	var range = document.getElementById("brush").value;
+```
+Let's break this down:
+- The first variable, uses the method `document.getElementById` to get the element from the html document with the specified `id`.
+- The second variable, takes the first variable and gets the canvas context using the method `getContext`. The canvas context provides the 2D rendering context for the drawing surface of a `<canvas>` element.
+- The third variable, takes the first variable and uses the method `getBoundingClientRect()`, which returns a DOMRect object providing information about the size of an element and its position relative to the viewport.
+- The fourth variable gets the element in the document with the id `brush` and gets the element's value using the method `value`.
+
+We need to create another 4 variables, but this time we will use these to configure some stuff for the canvas.
+```js
+	var mouseX = 0;
+	var mouseY = 0;
+	var isDrawing = false;
+	context.strokeStyle = 'black';
+```
+Let's break this down:
+- The first two variables, set the default position of the mouse with the X and Y coordinates.
+- The third variable will be used later to tell the application if the mouse is drawing something on the canvas. 
+- The `context.strokeStyle` property of the Canvas 2D API specifies the color, gradient, or pattern to use for the strokes (outlines) around shapes.
+
+Now, we'll need to handle the brush's size, in this block of code we’ll make sure the line width changes every time the input's value changes.
+```js
+//Handle brush size
+var brush = document.getElementById('brush');
+
+brush.addEventListener('input', function(brush){
+ context.lineWidth = brush.target.value;
+});
+```
+- First, we are creating a variable that uses the method `getElementById` to get an element from the document with the specified id. It gets the input element from the document.
+- The method `addEventListener` takes the brush element, and every time the value of this element changes, a function is called
+- This function takes the context of the canvas element and using the property `lineWidth`, it sets the thickness of the brush to the value from the range input.
+
+Then, we'll need to handle the colors of the brush, in this block of code we'll make sure that whenever the user clicks on any of the color buttons, the color of the brush changes.
+```js
+// Handle Colors
+var colors = document.getElementsByClassName('colors')[0];
+
+colors.addEventListener('click', function(event) {
+ context.strokeStyle = event.target.value || 'black';
+});
+```
+- First, we are creating a variable that uses the method `getElementByClassName` to get an element from the document with the specified class. It gets the `div` element from the document, this div container acts like an array and every element inside of it is indexed.
```

Comment:
i believe its not a single element which is returned and instead many elements are returned right?

in that case, `to get an element from the document with the specified class` is grammatically incorrect, i think